ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamApi.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.client.api;
+
+/**
+ * Stream data asynchronously to all the servers in the {@link org.apache.ratis.protocol.RaftGroup}.
+ * Clients may stream data to the nearest server and then the server will forward the data to the other servers.
+ * Once all the servers have received all the data of a request,
+ * the leader (may or may not be the nearest server) creates a log entry with a data ID generated by the state machine.
+ * Then, the leader sends the log entry to the followers.
+ * Since the followers already have received the data from the stream,
+ * they may lookup the data from the ID.
```

Comment:
Because follower needs to look up the data from the ID, so when follower receives the data from the stream, it needs to store a map<ID, the data from the stream>, so this ID must be transfered to all servers in the first request of stream.
Can this ID generated by clientId_streamId ?

Reply:
It depends on the StateMachine implementations.  They may use clientId_streamId as the data ID and choose to generate the ID in some other way.  For example, in Ozone, it may just use the Block/Chunk ID as the data ID.

Reply:
Thanks for explanation, Got it.